arch/common/gen_isr_tables.py

```diff
@@ -125,6 +125,45 @@ def parse_args():
 
     args = parser.parse_args()
 
+source_assembly_header = """
+#ifndef ARCH_IRQ_VECTOR_JUMP_CODE
+#define ARCH_IRQ_VECTOR_JUMP_CODE(v) "j " STRINGIFY(v)
+#endif
+"""
+
+def get_symbol_from_addr(syms, addr):
+    for key, value in syms.items():
+        if addr == value:
+            return key
+    return None
+
+def write_assembly_irq_vector_table(fp, vt, nv, syms):
```

Comment:
this "assembly" vs "c" nomenclature is quite ambiguous and does not align with the names used in the configs (i.e. "code" vs "address").

```suggestion
def write_code_irq_vector_table(fp, vt, nv, syms):
```